fixieai/agents/code_shot.py

```diff
@@ -97,6 +105,9 @@ def __init__(
         corpora: Optional[List[corpora.DocumentCorpus]] = None,
         conversational: bool = False,
         oauth_params: Optional[oauth.OAuthParams] = None,
+        model: Optional[str] = None,
```

Comment:
I think this ctor should get a `LlmSetting` object directly

Reply:
That was the route I went with initially but worried it made setting these too verbose -- but since these are advanced settings anyway I think it's okay.